plugins/solidigm/solidigm-telemetry/telemetry-log.h

```diff
@@ -11,10 +11,23 @@
 #include "libnvme.h"
 #include "util/json.h"
 
+#define SOLIDIGM_LOG_WARNING(...) fprintf(stderr, __VA_ARGS__); fprintf(stderr, "\n")
+
+#define CONCATENATE_VAR_NAMES(pre,post) pre ## post
+#define EVALUATE_CONCATENATE_VAR_NAMES(pre,post) CONCATENATE_VAR_NAMES(pre,post)
+#define STATIC_ASSERT(cond,msg) \
+    typedef struct { int CONCATENATE_VAR_NAMES(static_assertion_failed__,msg) : !!(cond); } \
+        EVALUATE_CONCATENATE_VAR_NAMES(static_assertion_failed_,__COUNTER__)
+
+#define MEMBER_SIZE(type, member) sizeof(((type *)0)->member)
```

Comment:
I'd like to avoid this macro madness. What about exploiding the fact we are using C99 which allows to use `_Static_assert`?

```c
#include <assert.h>

#if !defined __cplusplus
#define static_assert _Static_assert
#endif

static_assert(sizeof(unsigned int) == 4, "no scope");

int main(int argc, char *argv[])
{
	static_assert(sizeof(unsigned int) == 4, "inside main");
	return 0;
}
```

And then write this just as 

```c
static_assert(sizeof(struct reason_indentifier_1_0) == member_size(...,...), "size miss match");
```

and the `member_size` macro should definitely be in utils.h